### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by elasfarc**
In this PR; I did the following : 
- create `Classroom`, `Book` and Rental `classes`.
- Create the has-many/belongs-to relationship between `Classroom` and `Student`
- Create the many-to-many relationship between `Person` and `Book` through `Rental`

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  attr_accessor :label
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students.push(student)
```
Comment (arslanbisharat) [inline classroom.rb:11]: - [ ] Let's modify this method so it serves as the 'has-many side by replacing  `@students.push(student)` to 
`@students << student`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title:, author:)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (arslanbisharat) [inline book.rb]: - [ ] Let's add a method for adding rentals to book.rb. 
```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: classroom.rb (hunk 1 above)
Comment (elasfarc) [reply classroom.rb:11]: Screen Shot 2021-10-07 at 11 13 31 AM

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Review by SpaYco — COMMENTED** [review]
Hello 👋,

_This is an additional review as a result of the second opinion request._

You're right, using `<<` and `push` is fine as long as the code works, @arslanbisharat please take note!

I see that the project got approved though, great job staying professional!

Keep rocking 🚀!

**Pull request merged**